parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -110,6 +131,11 @@ public void writeTo(OutputStream out) throws IOException {
     }
 
     @Override
+    public byte[] getBytesUnsafe() {
+      return getBytes();
```

Comment:
This should add a comment to explain why this always copies the bytes: because the backing array is larger than the slice used for this Binary.